### ISSUE TITLE
ake stock price of any company you
want and predicts its price by using LSTM
Use only Jupyter notebook code.

### ISSUE THREAD

**Issue by Yarramvaramanikanta** [Yarramvaramanikanta/BharatIntern#1]
# Import necessary libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.preprocessing import MinMaxScaler
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import LSTM, Dense

# Load historical stock price data
# Replace 'your_data.csv' with your dataset
data = pd.read_csv('your_data.csv')
prices = data['Close'].values.reshape(-1, 1)

# Normalize the data
scaler = MinMaxScaler()
prices = scaler.fit_transform(prices)

# Create sequences and labels for LSTM training
seq_length = 20  # You can adjust this window size
X = []
y = []
for i in range(len(prices) - seq_length):
    X.append(prices[i:i+seq_length])
    y.append(prices[i+seq_length])

X = np.array(X)
y = np.array(y)

# Split data into training and testing sets
split_ratio = 0.8
split_index = int(split_ratio * len(X))
X_train, X_test = X[:split_index], X[split_index:]
y_train, y_test = y[:split_index], y[split_index:]

# Build an LSTM model
model = Sequential()
model.add(LSTM(units=50, activation='relu', input_shape=(seq_length, 1)))
model.add(Dense(units=1))
model.compile(optimizer='adam', loss='mean_squared_error')

# Train the model
model.fit(X_train, y_train, epochs=100, batch_size=32)

# Make predictions
predicted_prices = model.predict(X_test)

# Inverse transform the predictions to get actual prices
predicted_prices = scaler.inverse_transform(predicted_prices)
y_test = scaler.inverse_transform(y_test)

# Visualize the results
plt.figure(figsize=(12, 6))
plt.plot(y_test, label='True Prices', color='blue')
plt.plot(predicted_prices, label='Predicted Prices', color='red')
plt.legend()
plt.show()
